Resources/Computing/ARCComputingElement.py

```diff
@@ -180,6 +181,9 @@ def submitJob( self, executableFile, proxy, numberOfJobs = 1 ):
     """ Method to submit job
     """
 
+    # Assume that the ARC queues are always of the format nordugrid-<batchSystem>-<queue>
+    # And none of our supported batch systems have a "-" in their name
+    self.arcQueue = self.queue.split("-",2)[2]
```

Comment:
I would say that unless you are 100% sure that you have at least two "-" in the name, you should use try/except yes. And if you are sure, then put in the comment :-)